### PR TITLE
Create maximum_subsequence.py

### PR DESCRIPTION
**Pull request by pronoym99**
### Describe your change:
Add an algorithm for the ```maximum_subsequence_sum``` in an array inline with the existing ```maximum_subarray``` in the [other](https://github.com/TheAlgorithms/Python/tree/master/other) directory.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/maximum_subsequence.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Raises:
+      ValueError: when nums is empty.
+
+    >>> max_subsequence_sum([1,2,3,4,-2])
+    10
+    >>> max_subsequence_sum([-2, -3, -1, -4, -6])
+    -1
+    """
+    if not nums:
+        raise ValueError("Input sequence should not be empty")
```
Comment (CaedenPH) [inline other/maximum_subsequence.py:16]: This is kind of implied, but if you want to have this check, create a doctest that fails with it
ie 
```py
>>> max_subsequence_sum()
Traceback (most recent call last):
    ....
ValueError: Input sequence should not be empty
```

File: other/maximum_subsequence.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    ans = nums[0]
+    nums_len = len(nums)
+
+    for i in range(1, nums_len):
```
[the remark this answers is not in the document]
Comment (pronoym99) [reply other/maximum_subsequence.py:21]: ```len(nums)``` is computed for every ```i``` while we know that the length will remain constant throughout. Saving it in a variable speeds up the loop by roughly $33$ percent in large-sized ```nums```.

File: other/maximum_subsequence.py (hunk 1 above)
Comment (pronoym99) [reply other/maximum_subsequence.py:16]: Ok, I will do so.

**Comment by pronoym99** [comment]
@CaedenPH Please have a look at the updated code. It is pending for a long time. I have also added appropriate doctests as requested.

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**